src/views/forms/AccountManagementFormsNew/SlippageEditor.tsx

```diff
@@ -0,0 +1,153 @@
+import { useEffect, useRef, useState, type Dispatch, type SetStateAction } from 'react';
+
+import type { NumberFormatValues } from 'react-number-format';
+import styled from 'styled-components';
+
+import { ButtonShape, ButtonSize } from '@/constants/buttons';
+import { STRING_KEYS } from '@/constants/localization';
+
+import { useStringGetter } from '@/hooks/useStringGetter';
+
+import { FormInput } from '@/components/FormInput';
+import { Icon, IconName } from '@/components/Icon';
+import { InputType } from '@/components/Input';
+import { Output, OutputType } from '@/components/Output';
+import { ToggleGroup } from '@/components/ToggleGroup';
+import { WithConfirmationPopover } from '@/components/WithConfirmationPopover';
+
+enum EditorState {
+  Viewing = 'Viewing',
+  Selecting = 'Selecting',
+  Editing = 'Editing',
+}
+
+type ElementProps = {
+  slippage: number;
+  setIsEditing?: Dispatch<SetStateAction<boolean>>;
+  setSlippage: (slippage: number) => void;
+  disabled?: boolean;
+};
+
+export type SlippageEditorProps = ElementProps;
+
+export const SlippageEditor = ({
+  disabled,
+  slippage,
+  setIsEditing,
+  setSlippage,
+}: SlippageEditorProps) => {
+  const percentSlippage = slippage * 100;
+  const [slippageInputValue, setSlippageInputValue] = useState(percentSlippage.toString());
+  const [editorState, setEditorState] = useState(EditorState.Viewing);
+
+  const stringGetter = useStringGetter();
+
+  const toggleGroupRef = useRef<HTMLDivElement>(null);
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    setIsEditing?.(editorState !== EditorState.Viewing);
+
+    if (editorState === EditorState.Selecting) {
+      // use setTimeout with a 0ms delay to focus asynchronously.
+      setTimeout(() => toggleGroupRef?.current?.focus(), 0);
```

Comment:
```suggestion
      setImmediate(() => toggleGroupRef?.current?.focus());
```

better comment here would be to explain _why_ this has to happen asynchronously vs just explaining _what_ this line is doing

Reply:
hmm good point, let me find who originally made this

Reply:
makes sense since `toggleGroupRef.current.focus()` isn't a computationally expensive thing to do - so no need to queue it in separate processor thread